spec.bs

```diff
@@ -1721,14 +1721,14 @@ To <dfn>convert a modifier to a string</dfn> given a [=part/modifier=] |modifier
   1. If |init|["{{URLPatternInit/baseURL}}"] is not null:
     1. Set |baseURL| to the result of [=URL parser|parsing=] |init|["{{URLPatternInit/baseURL}}"].
     1. If |baseURL| is failure, then throw a {{TypeError}}.
-    1. Set |result|["{{URLPatternInit/protocol}}"] to |baseURL|'s [=url/scheme=].
-    1. Set |result|["{{URLPatternInit/username}}"] to |baseURL|'s [=url/username=].
-    1. Set |result|["{{URLPatternInit/password}}"] to |baseURL|'s [=url/password=].
-    1. Set |result|["{{URLPatternInit/hostname}}"] to |baseURL|'s [=url/host=] or the empty string if the value is null.
-    1. Set |result|["{{URLPatternInit/port}}"] to |baseURL|'s [=url/port=] or the empty string if the value is null.
-    1. Set |result|["{{URLPatternInit/pathname}}"] to the result of [=URL path serializing=] |baseURL|.
-    1. Set |result|["{{URLPatternInit/search}}"] to |baseURL|'s [=url/query=] or the empty string if the value is null.
-    1. Set |result|["{{URLPatternInit/hash}}"] to |baseURL|'s [=url/fragment=] or the empty string if the value is null.
+    1. Set |result|["{{URLPatternInit/protocol}}"] to the result of running [=process a base URL string=] given |baseURL|'s [=url/scheme=] and |type|.
+    1. Set |result|["{{URLPatternInit/username}}"] to the result of running [=process a base URL string=] given |baseURL|'s [=url/username=] and |type|.
+    1. Set |result|["{{URLPatternInit/password}}"] to the result of running [=process a base URL string=] given |baseURL|'s [=url/password=] and |type|.
+    1. Set |result|["{{URLPatternInit/hostname}}"] to the result of running [=process a base URL string=] given |baseURL|'s [=url/host=] and |type| or the empty string if the value is null.
```

Comment:
I think these sentences got a bit confusing. It's no longer clear what "or the empty string" attaches to.

Maybe "process a base URL string" could return the empty string if its first argument is null?

Reply:
That is a little different since previously we did not always check for null, but I think its ok to do this for base URL.  Alternatively, we could assert the component is not null because I believe URL parsing will not produce a null component value.

Anyway, I moved the check to the algorithm for now.

Reply:
Actually, I moved it to an assert.